var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -115,14 +131,23 @@ def install(self, spec, prefix):
                 'METIS_OPT=-I%s' % spec['metis'].prefix.include,
                 'METIS_LIB=%s' % metis_lib])
 
-        if '+mpi' in spec:
+        if 'mpi' in spec:
```

Comment:
This should be `+mpi`. Otherwise it will always run this block because `+mpi` and `~mpi` both match `mpi`. Same for `+parmetis` and `+metis` above.